code/datums/components/full_auto_fire.dm

```diff
@@ -281,7 +285,7 @@
 		if(-1 to 0)
 			target = get_step(shooter, shooter.dir) //Shoot in the direction faced if the mouse is on the same tile as we are.
 			target_loc = target
-		if(8 to INFINITY) //Can technically only go as far as 127 right now.
+		if(127 to INFINITY) //Can technically only go as far as 127 right now.
```

Comment:
We'll have to remove the `switch()` here to preserve the check.

Store the result of `get_dist(shooter, target)` in a variable, check if smaller or equal to zero, and do the stuff inside the `if(-1 to 0)` then.
And after that `else if(in_view_range(shooter, target))`